aspnet/fundamentals/file-providers/sample/src/FileProviderSample/Controllers/HomeController.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.FileProviders;
+
+namespace FileProviderSample.Controllers
+{
+    public class HomeController : Controller
+    {
+        private readonly IFileProvider _fileProvider;
+        private static string _dateFileLastChanged = "Never";
+
+        public HomeController(IFileProvider fileProvider)
+        {
+            _fileProvider = fileProvider;
+        }
+
+        public IActionResult Index()
+        {
+            var contents = _fileProvider.GetDirectoryContents("");
+            return View(contents);
+        }
+
+        public IActionResult Watch()
```

Comment:
Do you recommend having me just add a console app to the sample solution to demonstrate Watch? I'll start down that path.